sky/serve/infra_providers.py

```diff
@@ -622,10 +637,35 @@ def _probe_replica(info: ReplicaInfo) -> Tuple[str, bool]:
             if info.first_not_ready_time is None:
                 info.first_not_ready_time = time.time()
             if info.status_property.service_once_ready:
+                if self.use_spot:
```

Comment:
Why is it inside `if info.status_property.service_once_ready`? What if the cluster is preempted when the replica is in `STARTING` status?

Reply:
Also, do we need to refresh the cluster status every time the readiness probe fails? It might introduce some overhead here... We probably could only refresh when the consecutive failure threshold is exceeded, though that might lead to more downtime. cc @infwinston to take a look 🤔

Reply:
Re: inside of `service_once_ready`, I may have misunderstood the current code. If `service_once_ready == False`, is it possible that the VM just isn't up yet? If so, we won't be able to tell the difference between a VM not yet being up and a VM that was preempted, right?

Re: refresh cluster status - it's definitely not necessary, but like you said it will have an effect on downtime. How often do readiness probes fail in a healthy service? Is it actually often enough for refreshing cluster status to cause overhead?

Reply:
We will only probe replicas that have already finished their `sky launch` process. Please refer to the implementation of `should_track_status` 👀
https://github.com/skypilot-org/skypilot/blob/6981806fac2fe95866e7473575c3cb5bcd9d481a/sky/serve/infra_providers.py#L606-L610
That also reminds me that we won't probe replicas with a failed user job as well. But it is worth discussion whether we should restart such a replica since it might contain some user bug...

Reply:
Re: refresh cluster status:
Make sense. Current implementation looks great to me ; )

Reply:
Thanks for the pointer to `should_track_status`, makes sense! I moved the the code block outside of `if info.status_property.service_once_ready`